src/main/java/aeneas/models/Board.java

```diff
@@ -112,16 +116,40 @@ public boolean intersects(PlacedPiece piece) {
 
     return false;
   }
-  
+
   /**
    * Gets all the pieces currently on the board
    * @return The list of pieces on the board.
    */
   public ArrayList<PlacedPiece> getPieces() { return pieces; }
-  
+
   /**
    * Gets the list of hints currently on the board
    * @return The list of hints on the board.
    */
   public ArrayList<PlacedPiece> getHints() { return hints; }
+
+  /**
+   * Gets the state of all squares in the board. A square will be null if no square is there
+   * @return A two dimensional array representing the current state of the board
+   */
+  public Square[][] getSquares(){
+    Square[][] squares = new Square[SIZE][SIZE];
+    for (PlacedPiece piece : pieces){
+      for(Square s : piece.getSquaresInBoardFrame())
+        squares[s.getCol()][s.getRow()] = s;
```

Comment:
getSquaresInBoardFrame() already returns copies of the piece's squares, so there is no need to clone the squares again